src/azul/collections.py

```diff
@@ -57,23 +57,28 @@ class deep_dict_merge[K, V](dict):
     Recursively merge the given dictionaries. If more than one dictionary
     contains a given key, and all values associated with this key are themselves
     dictionaries, then the value present in the result is the recursive merging
-    of those nested dictionaries.
+    of those nested dictionaries. If all the values for a given key are not
+    dictionaries, an exception is raised unless `override` is true, in which
+    case the result will contain the latter of these values.
```

Comment:
```suggestion
    of those nested dictionaries. If none of the values for a given key are
    dictionaries, an exception is raised unless `override` is true, in which
    case the result will contain the latter of these values.
```

And what happens if some are dicts and some are not? Doctest is missing for this case, too.